.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,43 @@
+name: Dependabot on pre-commit hooks
+
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+
+on:
+  schedule:
+    - cron: 0 0 * * 0
```

Comment:
can you add a comment explaining the frequency? also there is a general advice when it comes to run cron "every day", it is to pick a random moment during the night instead of running it at midnight, you can just hardcode that "random" moment